### PR TITLE
Improve SdcaLogisticRegressionBinaryTrainer documentation.

### LINKED ISSUE #3501: SdcaLogisticRegressionBinaryTrainer documentation is hard to read
The discussion of algorithm and regularization are mixed into a big paragraph. It'd nicer to split them into different sections.

image

### PR DESCRIPTION
**Pull request by codemzs**
fixes  #3501

### CODE REVIEW

File: docs/api-reference/algo-details-sdca.md [hunk 1]
```diff
@@ -26,24 +26,27 @@ tight enough. In strongly-convex optimization, the optimal solution is unique
...
+This learner supports [elastic net
 regularization](https://en.wikipedia.org/wiki/Elastic_net_regularization): a
 linear combination of L1-norm (LASSO), $|| \textbf{w} ||_1$, and L2-norm
-(ridge), $|| \textbf{w} ||_2^2$ regularizations. L1-nrom and L2-norm
-regularizations have different effects and uses that are complementary in
-certain respects. Using L1-norm can increase sparsity of the trained
-$\textbf{w}$. When working with high-dimensional data, it shrinks small
+(ridge), $|| \textbf{w} ||_2^2$ regularizations. It can be specified by the
+'L2Const' and 'L1Threshold' parameters. Note that the 'L2Const' has an effect on
+the rate of convergence. In general, the larger the 'L2Const', the faster SDCA
```
Comment (wschin) [inline docs/api-reference/algo-details-sdca.md]: What does `converges` mean?

Comment (codemzs) [reply docs/api-reference/algo-details-sdca.md]: is there something you can suggest?

File: docs/api-reference/algo-details-sdca.md [hunk 2]
```diff
@@ -26,24 +26,27 @@ tight enough. In strongly-convex optimization, the optimal solution is unique
...
-selecting the optimal complexity in the bias-variance tradeoff. Regularization
-works by adding the penalty that is associated with coefficient values to the
-error of the hypothesis. An accurate model with extreme coefficient values would
-be penalized more, but a less accurate model with more conservative values would
-be penalized less. This learner supports [elastic net
+'NumThreads' to 1.
+
+This learner supports [elastic net
 regularization](https://en.wikipedia.org/wiki/Elastic_net_regularization): a
 linear combination of L1-norm (LASSO), $|| \textbf{w} ||_1$, and L2-norm
```
Comment (yaeldekel) [inline docs/api-reference/algo-details-sdca.md]: >L1-norm (LASSO), $|| \textbf{w} ||_1$, and L2-norm [](start = 22, length = 50)

I would change this to the same as what the Wikipedia page says:
```
a linear combination of the L1 and L2 penalties of the lasso and ridge methods
```
maybe with links for lasso and ridge.

File: docs/api-reference/algo-details-sdca.md [hunk 3]
```diff
@@ -26,24 +26,27 @@ tight enough. In strongly-convex optimization, the optimal solution is unique
...
-(ridge), $|| \textbf{w} ||_2^2$ regularizations. L1-nrom and L2-norm
-regularizations have different effects and uses that are complementary in
-certain respects. Using L1-norm can increase sparsity of the trained
-$\textbf{w}$. When working with high-dimensional data, it shrinks small
+(ridge), $|| \textbf{w} ||_2^2$ regularizations. It can be specified by the
+'L2Const' and 'L1Threshold' parameters. Note that the 'L2Const' has an effect on
+the rate of convergence. In general, the larger the 'L2Const', the faster SDCA
+converges. Regularization is a method that can render an ill-posed problem more
```
Comment (wschin) [inline docs/api-reference/algo-details-sdca.md]: the less number of iterations is needed to achieve a reasonable solution.

File: docs/api-reference/algo-details-sdca.md [hunk 4]
```diff
@@ -26,24 +26,27 @@ tight enough. In strongly-convex optimization, the optimal solution is unique
...
+tractable by imposing constraints that provide information to supplement the
+data and that prevents overfitting by penalizing model's magnitude usually
+measured by some norm functions. This can improve the generalization of the
```
Comment (wschin) [inline docs/api-reference/algo-details-sdca.md]: Regularization is a method that can render an ill-posed problem more
tractable and prevents overfitting by penalizing model's magnitude usually
measured by some norm functions.

File: docs/api-reference/algo-details-sdca.md [hunk 5]
```diff
@@ -26,24 +26,27 @@ tight enough. In strongly-convex optimization, the optimal solution is unique
...
+model learned by selecting the optimal complexity in the bias-variance tradeoff.
```
Comment (wschin) [inline docs/api-reference/algo-details-sdca.md]: >bias-variance tradeoff [](start = 57, length = 22)

[bias-variance tradeoff](https://en.wikipedia.org/wiki/Bias%E2%80%93variance_tradeoff)

File: docs/api-reference/algo-details-sdca.md (hunk 1 above)
Comment (wschin) [inline docs/api-reference/algo-details-sdca.md]: >rate of convergence [](start = 4, length = 19)

number of needed training iterations

**Pull request merged**